controllers/capabilities/activedoc_threescale_reconciler.go

```diff
@@ -0,0 +1,302 @@
+package controllers
+
+import (
+	"errors"
+	"net/url"
+	"reflect"
+
+	capabilitiesv1beta1 "github.com/3scale/3scale-operator/apis/capabilities/v1beta1"
+	controllerhelper "github.com/3scale/3scale-operator/pkg/controller/helper"
+	"github.com/3scale/3scale-operator/pkg/helper"
+	"github.com/3scale/3scale-operator/pkg/reconcilers"
+
+	threescaleapi "github.com/3scale/3scale-porta-go-client/client"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+type ActiveDocThreescaleReconciler struct {
+	*reconcilers.BaseReconciler
+	resource            *capabilitiesv1beta1.ActiveDoc
+	threescaleAPIClient *threescaleapi.ThreeScaleClient
+	providerAccountHost string
+	logger              logr.Logger
+}
+
+func NewActiveDocThreescaleReconciler(b *reconcilers.BaseReconciler, resource *capabilitiesv1beta1.ActiveDoc, threescaleAPIClient *threescaleapi.ThreeScaleClient, providerAccountHost string, logger logr.Logger) *ActiveDocThreescaleReconciler {
+	return &ActiveDocThreescaleReconciler{
+		BaseReconciler:      b,
+		resource:            resource,
+		threescaleAPIClient: threescaleAPIClient,
+		providerAccountHost: providerAccountHost,
+		logger:              logger.WithValues("3scale Reconciler", providerAccountHost),
+	}
+}
+
+func (s *ActiveDocThreescaleReconciler) Reconcile() (*threescaleapi.ActiveDoc, error) {
+	s.logger.V(1).Info("START")
+
+	remoteActiveDocs, err := s.threescaleAPIClient.ListActiveDocs()
+	if err != nil {
+		return nil, err
+	}
+
+	var remoteActiveDoc *threescaleapi.ActiveDoc
+
+	for idx := range remoteActiveDocs.ActiveDocs {
```

Comment:
Isn't there a way to get the desired ActiveDoc directly instead of listing them?

Reply:
Only by ID. But we use the systemName as the key identifier. If you change the systemname of the CR, the CR will reconcile a 3scale remote object with that systemname, instead of changing the systemname of a existing 3scale remote object (which cannot be done)